ndcube/ndcube.py

```diff
@@ -1461,6 +1461,59 @@ def fill_masked(self, fill_value, uncertainty_fill_value=None, unmask=False, fil
             self.mask = False
         return None
 
+    def to_nddata(self, *, nddata_type=NDData, **kwargs):
+        """
+        Constructs new type instance with the same attribute values as this `~ndcube.NDCube`.
+
+        Any attributes not supported by the new class (``nddata_type``), will be discarded.
+        Values of supported attributes can be altered on the output object by setting a
+        kwarg with the new value, e.g. ``data=new_data``.
+
+        Parameters
+        ----------
+        nddata_type:
+            The type of the returned object. Must be a subclass of `~astropy.nddata.NDData`
+            or a class that behaves like one.  Default=`~astropy.nddata.NDData`.
+
+        kwargs:
+            Attributes to change on output object. For example, to change the data on the
+            returned object compare to this instance, set a kwarg ``data=new_data``.
+            Note that kwargs set to ``None`` will not be passed to the constructor
+            of the ``nddata_type``. Therefore, if that attribute exists on the new type,
+            it will be given the default value defined by ``nddata_type``'s call signature.
```

Comment:
```suggestion
            Inputs to the ``nddata_type`` constructor that should differ from the values
            stored in attributes of this instance. For example, to set different data values
            on the returned object, set a kwarg ``data=new_data``, where ``new_data`` is an
            an array of compatible shape and dtype.  Note that kwargs set to ``None`` will
            not be passed to the constructor of the ``nddata_type``. Therefore, if that
            attribute exists on the new type, it will be given the default value defined by
            ``nddata_type``'s call signature.  Note that kwargs given by the user and
            attributes on this instance that are not supported by the ``nddata_type``
            constructor are ignored.
```